examples/psbt.rs

```diff
@@ -0,0 +1,360 @@
+    }
+
+    /// Returns the fingerprint for the master extended public key.
+    fn master_fingerprint(&self) -> Fingerprint {
+        self.master_xpub.fingerprint()
+    }
+
+    /// Signs `psbt` with this signer.
+    fn sign_psbt<C: Signing>(&self, secp: &Secp256k1<C>, mut psbt: Psbt) -> Result<Psbt> {
+        psbt.sign(self.master_xpriv, secp)?;
+        Ok(psbt)
+    }
+}
+
+/// An example of an watch-only online wallet.
+struct WatchOnly {
+    /// The xpub for account 0 derived from derivation path "m/84h/0h/0h".
+    account_0_xpub: ExtendedPubKey,
+    /// The xpub derived from `INPUT_UTXO_DERIVATION_PATH`.
+    input_xpub: ExtendedPubKey,
+    /// The master extended pubkey fingerprint.
+    master_fingerprint: Fingerprint,
+}
+
+impl WatchOnly {
+    /// Constructs a new watch-only wallet.
+    ///
+    /// A watch-only wallet would typically be online and connected to the Bitcoin network. We
+    /// 'import' into the wallet the `account_0_xpub` and `master_fingerprint`.
+    ///
+    /// The reason for importing the `input_xpub` is so one can use bitcoind to grab a valid input
+    /// to verify the workflow presented in this file.
+    fn new(account_0_xpub: ExtendedPubKey, input_xpub: ExtendedPubKey, master_fingerprint: Fingerprint) -> Self {
+        WatchOnly { account_0_xpub, input_xpub, master_fingerprint }
+    }
+
+    /// Creates the PSBT, in BIP174 parlance this is the 'Creater'.
+    fn create_psbt<C: Verification>(&self, secp: &Secp256k1<C>) -> Result<Psbt> {
+        let to_address = Address::from_str(RECEIVE_ADDRESS)?;
+        let to_amount = Amount::from_str(OUTPUT_AMOUNT_BTC)?;
+
+        let (_, change_address, _) = self.change_address(secp)?;
+        let change_amount = Amount::from_str(CHANGE_AMOUNT_BTC)?;
+
+        let tx = Transaction {
+            version: 2,
+            lock_time: 0,
+            input: vec![
+                TxIn {
+                    previous_output: OutPoint {
+                        txid: Txid::from_hex(INPUT_UTXO_TXID)?,
+                        vout: INPUT_UTXO_VOUT,
+                    },
+                    script_sig: Script::new(),
+                    sequence: 0xFFFFFFFF, // Ignore nSequence.
+                    witness: Witness::default(),
+                },
+            ],
+            output: vec![
+                TxOut {
+                    value: to_amount.to_sat(),
+                    script_pubkey: to_address.script_pubkey(),
+                },
+                TxOut {
+                    value: change_amount.to_sat(),
+                    script_pubkey: change_address.script_pubkey(),
+                }
+            ],
+        };
+
+        let psbt = Psbt::from_unsigned_tx(tx)?;
+
+        Ok(psbt)
+    }
+
+    /// Updates the PSBT, in BIP174 parlance this is the 'Updater'.
+    fn update_psbt(&self, mut psbt: Psbt) -> Result<Psbt> {
+        let mut input = Input::default();
+
+        input.witness_utxo = Some(previous_output());
```

Comment:
We can clean this up on the new MSRV :)
```suggestion
        let mut input = Input { witness_utxo: Some(previous_output()), ..Default::default() };
```

Reply:
Done, cheers!